src/utils/io/mmap.rs

```diff
@@ -0,0 +1,37 @@
+// Copyright 2019-2023 ChainSafe Systems
+// SPDX-License-Identifier: Apache-2.0, MIT
+
+use std::io;
+
+use memmap2::MmapAsRawDesc;
+use positioned_io::{ReadAt, Size};
+
+/// Wrapper type of [`memmap2::Mmap`] that implements [`ReadAt`] and [`Size`]
+pub struct Mmap(memmap2::Mmap);
```

Comment:
Let's have some tests for this, there tend to be dragons in buffers and indexing.

Reply:
Done